opm/simulators/wells/MultisegmentWellPrimaryVariables.cpp

```diff
@@ -120,6 +129,7 @@ update(const WellState<Scalar>& well_state,
                 const int gas_pos = pu.phase_pos[Gas];
                 value_[seg][GFrac] = well_.scalingFactor(gas_pos) * segment_rates[well_.numPhases() * seg + gas_pos] / total_seg_rate;
             }
+            // what about water and gas injection?
```

Comment:
Just explaining the question in the comments, for other segments, they can use the same formula, for top segments, segment rates are the well rates, for water and gas injection, they will have a non-zero rates for either water or gas, while it does not break the formula either.

Reply:
If you agree with the response, you can remove the comment.